### DISCUSSION TITLE
Does it work with notifications?

### DISCUSSION THREAD

**Discussion by pauleniz** [Xammie/mailbook#27] [Q&A] [answered]
I'm trying to display notifications in mailbook but it expects instance of Illuminate\Contracts\Mail\Mailable is it possible to convert notification to mailable? I tried googling it but didn't find the answer.

**Comment by Xammie** [comment]
Currently notification are not supported. It is however an interesting feature which I may add or you can create a PR yourself.

**Comment by Xammie** [comment]
@pauleniz I've finished the support for notifications and would appreciate if you could test it. Let me know how it goes!

```
composer require xammie/mailbook dev-support-notifications
```

**Reply by TonyKrus** [reply]
I tested it and it work well with Notifications.

But with the version 1.4.1, I can use simple instance of my model but not with the dev-support-notification branch.

For example, this code works on 1.4.1 but not dev-support-notification
```php
Mailbook::add(OrderCreated::class)
    ->variant('Status A', function () {
        return new OrderCreated(Order::factory()->make());
    });
```

I get a 404.

**Reply by Xammie** [reply]
Thanks for the feedback, I will look into it.

**Reply by Xammie** [reply]
I'm not able to reproduce the issue with your example. Do you have any queries in your mailable that are throwing a `ModelNotFoundException` ? You could try to create the model.

```php
return new OrderCreated(Order::factory()->create());
```

Remember to enable database support: https://github.com/Xammie/mailbook#using-the-database

**Reply by TonyKrus** [reply]
Actually in my case, I have 2 in memory models, a "Thread" and a "Message".

Here is the exact code I have :
```php
Mailbook::add(ContactMail::class)
    ->label('Formulaire de contact')
    ->variant('Nouveau message', function () {
        $contact = Contact::factory()->make(['id' => 1]);
        $contactsMessage = ContactsMessage::factory()->make([
            'contact_id' => 1,
            'admin_id'   => 1,
        ]);

        return new ContactMail($contact, $contactsMessage, true);
    });
```

I don't want to persist model because we use the package in production.

**Reply by Xammie** [reply]
I understand that you wouldn't want to fill your database with random records, however, mailbook supports this by using database transactions https://github.com/Xammie/mailbook#using-the-database. 

It works by using database transactions. Any changes made to the database while the mail is rendering are going to be rolled back.

Still to solve your problem without using database rollback I would need more details about what your mailable is doing. Because I suspect that the mailable is maybe executing a query that fails.

**Comment by Xammie** [comment, marked as answer]
I just released version [1.5.0](https://github.com/Xammie/mailbook/releases/tag/1.5.0) which now supports notifications! 🎉
